### PR TITLE
osd, mgr, scripts: support 'deep_scrub' command format and use it in the relevant scripts

### PR DESCRIPTION
**Pull request by ronen-fr**
While 'deep-scrub' is the new command argument format (and
'deep_scrub' is deprecated, we must deal with mixed clusters.

This PR:
- restores the 'deep_scrub' command format in the osd (as
  a working but deprecated alternative);
- restore support for both formats in the manager;
- revert ceph.py to using 'deep_scrub' for compatibility
  with pre-Squid daemons;

### CODE REVIEW

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -4494,6 +4504,14 @@ void OSD::final_init()
     asok_hook,
     "Trigger a deep scrub");
   ceph_assert(r == 0);
+  // a deprecated form of the above. Required for
+  // upgrade compatibility & upgrade testing.
+  r = admin_socket->register_command(
+    "deep_scrub "
+    "name=pgid,type=CephPgid,req=false",
+    asok_hook,
+    "deprecated! use 'deep-scrub' instead");
```
Comment (rzarzynski) [inline src/osd/OSD.cc:4513]: ACK.

File: qa/tasks/ceph.py [hunk 2]
```diff
@@ -1271,7 +1271,8 @@ def osd_scrub_pgs(ctx, config):
         try:
             manager.raw_cluster_cmd('tell', 'osd.' + id_, 'config', 'set',
                                     'osd_debug_deep_scrub_sleep', '0');
-            manager.raw_cluster_cmd('osd', 'deep-scrub', id_)
+            # using the old (deprecated) format, catering for mixed clusters
+            manager.raw_cluster_cmd('osd', 'deep_scrub', id_)
```
[the remark this answers is not in the document]
Comment (rzarzynski) [reply qa/tasks/ceph.py:1275]: Yeah, keeping in mind the need for atomicity of retrieval of the cmd schema and issuing the actual command

The problem I'm more worried about is whether there is an external user (script) of the old command. Knocking it out without proper deprecation would be controversial.

[the remark this answers is not in the document]
Comment (ronen-fr) [reply qa/tasks/ceph.py:1275]: Thanks for the comments. I will change the PR as follows:
- keep supporting both formats in the OSD;
- allow both formats to 'pass thru' the manager;
- use the new command if available in the scripts, but modify the upgrade/mixed ones (ceph.py, ...) to fallback to the old if needed.

File: src/pybind/mgr/restful/common.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 # Implemented osd commands
 OSD_IMPLEMENTED_COMMANDS = [
-    'scrub', 'deep-scrub', 'repair'
+    'scrub', 'deep-scrub', 'deep_scrub', 'repair'
```
Comment (rzarzynski) [inline src/pybind/mgr/restful/common.py:9]: Was ever `deep_scrub` exposed here?

Comment (neha-ojha) [inline src/pybind/mgr/restful/common.py:9]: Why do we need to expose `deep_scrub` to the restful module? It was never exposed in earlier versions https://github.com/ceph/ceph/blob/quincy/src/pybind/mgr/restful/common.py#L9 and https://github.com/ceph/ceph/blob/reef/src/pybind/mgr/restful/common.py#L9?

Comment (ronen-fr) [reply src/pybind/mgr/restful/common.py:9]: yes - but (starting at some point in the past, when someone "fixed" this file) not with the common command format (tell [pg] deep-scrub).

Comment (ronen-fr) [reply src/pybind/mgr/restful/common.py:9]: But I agree with you and Neha's comment below: as this was not supported in the last two versions,
I will run the upgrade tests without it.

**Comment by ronen-fr** [comment]
closing for now. A minimal viable fix is proposed as https://github.com/ceph/ceph/pull/56745

**Pull request closed without merging**